internal/dispatch/remote/cluster.go

```diff
@@ -881,6 +964,46 @@ func (srst *supportedResourceSubjectTracker) updateForError(err error) {
 	}
 }
 
+type primarySleeper struct {
+	reqKey   string
+	waitTime time.Duration
+	cancel   context.CancelFunc // GUARDED_BY(lock)
+	lock     sync.Mutex
+}
+
+func (s *primarySleeper) sleep(ctx context.Context) {
+	if s.waitTime <= 0 {
+		return
+	}
+
+	s.lock.Lock()
+	ctx, s.cancel = context.WithCancel(ctx)
+	s.lock.Unlock()
```

Comment:
Instead of using a lock here, you could create the context on `newPrimarySleeper`, that way you'll know there is no possibility of races.

Reply:
If I do so, it is not resuable